docs/source/developer/distributions.rst

```diff
@@ -0,0 +1,232 @@
+   :scale: 100%
+
+Base submodule
+~~~~~~~~~~~~~~
+The base module contains the :class:`~metasyn.distribution.BaseDistribution` class, which is the base class for all distributions. It also contains the :class:`~metasyn.distribution.ScipyDistribution` class, which is a specialized base class for distributions that are built on top of SciPy's statistical distributions. 
+
+Additionally it contains the :class:`~metasyn.distribution.UniqueDistributionMixin` class, which is a mixin class that can be used to indicate that a distribution is unique (i.e., that it does not contain duplicate values).
+
+Finally it contains the :func:`~metasyn.distribution.metadist` decorator, which is used to set the attributes of a distribution (e.g., ``implements``, ``var_type``, etc.).
+
+BaseDistribution class
+^^^^^^^^^^^^^^^^^^^^^^
+This is the base class providing the basic structure for all distributions. It is not intended to be used directly, but rather to be derived from when implementing a new distribution.
+
+**Attributes:**
+
+- ``implements``: A unique string identifier for the distribution type, e.g. ``core.uniform`` or ``core.poisson``.
+- ``var_type``: The type of variable associated with the distribution, e.g. ``discrete`` or ``continuous``.
+- ``provenance``: Information about the source (core, plugin, etc.) of the distribution.
+- ``privacy``: The privacy class or implementation associated with the distribution.
+- ``is_unique``: A boolean indicating whether the values in the distribution are unique.
+- ``version``: The version of the distribution. 
+
+**Properties:**
+
+- :meth:`~metasyn.distribution.BaseDistribution._params_formatted`: Provides a formatted string of the distribution's parameters for easy readability.
+
+**Methods:**
+
+- :meth:`~metasyn.distribution.BaseDistribution.fit`: Class method to fit a distribution to a given series. 
+- :meth:`~metasyn.distribution.BaseDistribution._fit`: Abstract class method intended to contain the fitting logic for the distribution. It does not need to handle N/A values. **It must be implemented by derived classes.**
+- :meth:`~metasyn.distribution.BaseDistribution._to_series`: Static method converting different data types (Polars Series, Pandas Series, or sequences) into a Polars Series, handling null values appropriately.
+- :meth:`~metasyn.distribution.BaseDistribution.draw`: Abstract method, intended to draw a new value from the distribution. **It must be implemented by derived classes.**
+- :meth:`~metasyn.distribution.BaseDistribution.draw_reset`: Method to reset the distribution's drawing mechanism. This should be implemented if the drawing does not happen randomly.
+- :meth:`~metasyn.distribution.BaseDistribution._param_dict`: Abstract method to return a dictionary of the distribution's parameters. 
+- :meth:`~metasyn.distribution.BaseDistribution.to_dict`: Method to create a dictionary representation of the distribution. **It must be implemented by derived classes.**
```

Comment:
Ah, dang, I already had my suspicions, but the docstring in BaseDistribution specifically mentions these three. I'll make an issue to change the docstring to avoid confusion.